src/Compilers/Core/Portable/Binding/BindingDiagnosticBag.cs

```diff
@@ -99,17 +103,19 @@ internal void Clear()
             DependenciesBag?.Clear();
         }
 
-        internal void AddRange(ImmutableBindingDiagnostic<TAssemblySymbol> other)
+        internal void AddRange(ImmutableBindingDiagnostic<TAssemblySymbol> other, bool allowMismatchInDependencyAccumulation = false)
```

Comment:
>bool allowMismatchInDependencyAccumulation = false [](start = 82, length = 50)

In the past we've made these verification-only parameters debug only. Do we want to do that here as well?

Reply:
> In the past we've made these verification-only parameters debug only. Do we want to do that here as well?

VB doesn't allow conditional compilation in the middle of a statement. That is why I decided against this approach. If you are concerned with passing this argument, I could add specialized methods without this argument instead.

---
In reply to: [369807116](https://github.com/dotnet/roslyn/pull/41117#discussion_r369807116) [](ancestors = 369807116)

Reply:
> If you are concerned with passing this argument, I could add specialized methods without this argument instead.

I'm not particularly concerned, it was more of a question as to why you weren't doing it than it was a request to do so.

---
In reply to: [369816556](https://github.com/dotnet/roslyn/pull/41117#discussion_r369816556) [](ancestors = 369816556,369807116)